src/core/network/polling.ts

```diff
@@ -45,6 +48,10 @@ export class Polling {
           await wait(retryAfter * 1000)
           continue
         }
+        if (err instanceof TelegramError && err.code === 409) {
+          this.confirmUpdates = noop
```

Comment:
This makes it impossible to restart the bot after the error!

Also, arbitrarily rewriting instance members on the fly is generally really causing mindfuck bugs. The same solution can be achieved with a simple flag and not much more effort.

Reply:
`throw`ing here causes unhandled rejection (= process crash on Node 15, preventing any attempts at reusing `Polling`).

401 should probably also avoid confirming, since it's guaranteed to fail.

Reply:
`Polling` objects are meant to be recreated, not reused. To make reuse impossible, I can abort the controller in `finally`. It can't be replaced since it's `readonly`.

Reply:
I don't know how the member would need to be named so that the code is as clear as `this.confirmUpdates = noop`. It wouldn't be used for anything else.

Reply:
> causes unhandled rejection

does it really? It is awaited inside a loop, no?

Reply:
Yes, but `.loop` is not awaited.